sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -164,27 +164,33 @@ def _load_balancing(self):
         """
         while self._running:
             try:
-                checkpoints = self._ownership_manager.get_checkpoints() if self._checkpoint_store else None
                 claimed_partition_ids = self._ownership_manager.claim_ownership()
                 if claimed_partition_ids:
-                    to_cancel_list = set(self._consumers.keys()) - set(claimed_partition_ids)
-                    self._create_tasks_for_claimed_ownership(claimed_partition_ids, checkpoints)
+                    existing_pids = set(self._consumers.keys())
+                    claimed_pids = set(claimed_partition_ids)
+                    to_cancel_pids = existing_pids - claimed_pids
+                    newly_claimed_pids = claimed_pids - existing_pids
+                    if newly_claimed_pids:
+                        checkpoints = self._ownership_manager.get_checkpoints() if self._checkpoint_store else None
+                        self._create_tasks_for_claimed_ownership(newly_claimed_pids, checkpoints)
                 else:
                     _LOGGER.info("EventProcessor %r hasn't claimed an ownership. It keeps claiming.", self._id)
-                    to_cancel_list = set(self._consumers.keys())
-                if to_cancel_list:
-                    self._cancel_tasks_for_partitions(to_cancel_list)
+                    to_cancel_pids = set(self._consumers.keys())
+                if to_cancel_pids:
+                    self._cancel_tasks_for_partitions(to_cancel_pids)
             except Exception as err:  # pylint:disable=broad-except
                 _LOGGER.warning("An exception (%r) occurred during balancing and claiming ownership for "
                                 "eventhub %r consumer group %r. Retrying after %r seconds",
                                 err, self._eventhub_name, self._consumer_group, self._load_balancing_interval)
+                self._handle_callback(self._error_handler, None, err)
                 # ownership_manager.get_checkpoints() and ownership_manager.claim_ownership() may raise exceptions
                 # when there are load balancing and/or checkpointing (checkpoint_store isn't None).
                 # They're swallowed here to retry every self._load_balancing_interval seconds.
                 # Meanwhile this event processor won't lose the partitions it has claimed before.
                 # If it keeps failing, other EventProcessors will start to claim ownership of the partitions
                 # that this EventProcessor is working on. So two or multiple EventProcessors may be working
-                # on the same partition.
+                # on the same partition for a short while.
+                # When owner_levle by default is set, this time duration will be very short.
```

Comment:
typo: owner_level.

and I think the statement could be improved to like:
Setting owner_level would create exclusive connection to the partition which would alleviate duplicate-receiving greatly.